### PR TITLE
[ESIMD] Implement unified memory API - part2 - block_load(acc, ...)

### PR DESCRIPTION
**Pull request by v-klochkov**

### CODE REVIEW

File: sycl/include/sycl/ext/intel/esimd/memory.hpp [hunk 1]
```diff
@@ -266,127 +223,21 @@ scatter(Tx *p, Toffset offset, simd<Tx, N> vals, simd_mask<N> mask = 1) {
...
-                                    vals.data());
-}
-
-namespace detail {
-template <
-    typename T, int NElts,
-    __ESIMD_DNS::lsc_data_size DS = __ESIMD_DNS::lsc_data_size::default_size,
-    cache_hint L1H = cache_hint::none, cache_hint L2H = cache_hint::none,
-    typename FlagsT>
+template <typename T, int NElts, cache_hint L1H = cache_hint::none,
```
Comment (v-klochkov) [inline sycl/include/sycl/ext/intel/esimd/memory.hpp:235]: this block_load_impl() function is actually the copy of the old experimental::lsc_block_load().
Please pay attention that this function here doesn't have the 'lsc_data_size' template parameter that was used in the old experimental::lsc_block_load().
The reason for that is that the data-size is always deduced from the loaded element type.

File: sycl/include/sycl/ext/intel/esimd/memory.hpp [hunk 2]
```diff
@@ -266,127 +223,21 @@ scatter(Tx *p, Toffset offset, simd<Tx, N> vals, simd_mask<N> mask = 1) {
...
-/// @tparam N Number of elements to load.
-/// @tparam Flags The alignment specifier type tag.
-/// @param addr The address to load from.
-/// @param Flags Specifies the alignment.
-/// @return A vector of loaded elements.
-///
-template <typename Tx, int N,
-          typename Flags = overaligned_tag<detail::OperandSize::OWORD>>
-__ESIMD_API std::enable_if_t<is_simd_flag_type_v<Flags>, simd<Tx, N>>
-block_load(const Tx *addr, Flags) {
```
Comment (v-klochkov) [inline sycl/include/sycl/ext/intel/esimd/memory.hpp:287]: This function and the next 2 are not removed - they are just moved, grouped with other user-visible block_load() functions (and block_store() functions).

File: sycl/include/sycl/ext/intel/esimd/memory.hpp [hunk 3]
```diff
@@ -858,17 +928,376 @@ block_load(const T *ptr, simd_mask<1> pred, simd<T, N> old_values,
...
+///                      or 512(only if alignment is 8-bytes or more).
+/// R3: The target device must be DG2, PVC or newer GPU.
+/// (R1), (R2) and (R3) are not applied if there are no cache hints.
+template <typename T, int N, typename AccessorT,
+          typename PropertyListT =
+              ext::oneapi::experimental::detail::empty_properties_t>
+__ESIMD_API std::enable_if_t<
+    ext::oneapi::experimental::is_property_list_v<PropertyListT> &&
+        sycl::detail::acc_properties::is_accessor_v<AccessorT> &&
+        !sycl::detail::acc_properties::is_local_accessor_v<AccessorT>,
```
Comment (sarnex) [inline sycl/include/sycl/ext/intel/esimd/memory.hpp:1084]: Maybe a dumb question but we don't allow local accessors for a couple of the functions here, why is that?

File: sycl/test-e2e/ESIMD/unified_memory_api/block_load_acc_pvc.cpp [hunk 4]
```diff
@@ -0,0 +1,42 @@
+//==------- block_load_usm_pvc.cpp - DPC++ ESIMD on-device test ------------==//
```
Comment (sarnex) [inline sycl/test-e2e/ESIMD/unified_memory_api/block_load_acc_pvc.cpp:1]:
```suggestion
//==------- block_load_acc_pvc.cpp - DPC++ ESIMD on-device test ------------==//
```

**Review by sarnex — APPROVED** [review]
LGTM! Only minor comments!

File: sycl/include/sycl/ext/intel/esimd/memory.hpp (hunk 3 above)
Comment (v-klochkov) [reply sycl/include/sycl/ext/intel/esimd/memory.hpp:1084]: There is such function in experimental/memory.hpp. It is just not ported by this patch yet:
```
template <typename T, int NElts, lsc_data_size DS = lsc_data_size::default_size,
          cache_hint L1H = cache_hint::none, cache_hint L3H = cache_hint::none,
          typename AccessorTy,
          typename FlagsT = __ESIMD_DNS::dqword_element_aligned_tag>
__ESIMD_API std::enable_if_t<
    sycl::detail::acc_properties::is_local_accessor_v<AccessorTy> &&
        __ESIMD_NS::is_simd_flag_type_v<FlagsT>,
    __ESIMD_NS::simd<T, NElts>>
lsc_block_load(AccessorTy acc, uint32_t offset,
               __ESIMD_NS::simd_mask<1> pred = 1, FlagsT flags = FlagsT{}) {
  return lsc_slm_block_load<T, NElts, DS>(
      offset + __ESIMD_DNS::localAccessorToOffset(acc), pred);
}
```

**Comment by v-klochkov** [comment]
@sarnex - Thank you for quick code-review. Because the comments required only few NFC changes - I will merge the current change and upload a separate NFC fix now.

File: sycl/test-e2e/ESIMD/unified_memory_api/block_load_acc_pvc.cpp (hunk 4 above)
Comment (v-klochkov) [reply sycl/test-e2e/ESIMD/unified_memory_api/block_load_acc_pvc.cpp:1]: This and other comments are addressed in https://github.com/intel/llvm/pull/11594

File: sycl/include/sycl/ext/intel/esimd/detail/memory_intrin.hpp [hunk 5]
```diff
@@ -223,7 +281,7 @@ __ESIMD_INTRIN void __esimd_svm_scatter4_scaled(
 //   0 - 1 byte, 1 - 2 bytes, 2 - 4 bytes
 // @tparam Scale - offset scaling factor; must be zero currently
 // @tparam L1H - L1 cache hint
-// @tparam L3H - L3 cache hint
+// @tparam L2H - L2 cache hint
```
[the remark this answers is not in the document]
Comment (v-klochkov) [reply sycl/include/sycl/ext/intel/esimd/detail/memory_intrin.hpp:284]: Removed the whole function in https://github.com/intel/llvm/pull/11594

**Pull request merged**